spec/requests/auth0_controller_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Auth0Controller, type: :request do
+  describe 'GET /callback' do
+    it 'returns http success' do
+      user = { test: 'test' }
+      Utils::Auth.login(user, roles: [])
+      get auth_callback_path, params: { code: 'TEST' }
+      expect(response).to redirect_to(dashboard_path)
+    end
+  end
+
+  describe 'GET /failure' do
+    it 'returns http success' do
+      get auth_failure_path
+      expect(response).to have_http_status(:bad_request)
+    end
+  end
+
+  describe 'GET /logout' do
+    it 'returns http success' do
+      user = { test: 'test' }
+      Utils::Auth.login(user, roles: [])
```

Comment:
❗ After implementing `mock_user`, call it here:
```suggestion
      Utils::Auth.login(mock_user, roles: [])
```